ui/src/table.tsx

```diff
@@ -727,35 +728,50 @@ export const
         const expandedRef = expandedRefs[key]
         return expandedRef === undefined || expandedRef
       },
+      initialGroups = React.useMemo(() => {
+        return m.groups
+          ? m.groups.reduce((acc, { label }) => {
+            acc[label] = ({ key: label, name: label, startIndex: 0, count: 0 })
+            return acc
+          }, {} as Dict<Fluent.IGroup>)
+          : groupByKey !== '*'
+            ? items.reduce((acc, item) => {
+              const group = (item as Fluent.IObjectWithKey & Dict<any>)[groupByKey]
+              if (!acc[group]) acc[group] = ({ key: group, name: group, startIndex: 0, count: 0 })
+              return acc
+            }, {} as Dict<Fluent.IGroup>)
+            : {}
+      }, [m.groups, groupByKey, items]),
       makeGroups = React.useCallback((groupByKey: S, filteredItems: (Fluent.IObjectWithKey & Dict<any>)[]) => {
+        const allGroups: Dict<Fluent.IGroup> = { ...initialGroups }
```

Comment:
If group names are enough:

```suggestion
        const allGroups = [...groupNames].reduce((acc, groupName) => {
          acc[groupName] = { key: groupName, name: groupName, startIndex: 0, count: 0, isCollapsed: getIsCollapsed(groupName, expandedRefs.current) }
          return acc
        }, {} as Dict<Fluent.IGroup>)
```